File: workflow_shell/git/go.py
from ..cli import all_commands

command_string = "ggo"
command_help = (
    "Takes one optional argument, number of commits to rebase. "
    + "Commits all staged changes, "
    + "rebases interactively, "
    + "amends the most recent commit, "
    + "then force pushes"
)
short_help = "Git - commits all, rebases, amends, force pushes"
command_arguments = ["number_of_commits"]
argument_required = False
argument_default = 0


def main(*args, **kwargs):
    context = args[0]

    # Commit current changes
    context.invoke(all_commands["gcam"])

    # Interactive rebase
    argument = kwargs.get(command_arguments[0])
    if argument > 0:
        context.invoke(all_commands["grb"], number_of_commits=argument)

    # Amend commit
    context.invoke(all_commands["ga"])

    # Push
    context.invoke(all_commands["gpfo"])

File: src/commands/git/Go.ts
import { Option } from '../../common/interface/Option';
import { SuperCommand } from '../../common/SuperCommand';
import { executeCommand } from '../../common/Cli';
import { Argument } from '../../common/interface/Argument';
import { Command as GitCommit } from './CommitAllWithMessage';
import { Command as GitRebase } from './Rebase';
import { Command as GitAmend } from './Amend';
import { Command as GitForcePush } from './ForcePush';
import { Names } from '../../common/interface/Names';

const argumentss: Argument[] = [Argument.NUMBER];
const options: Option[] = [];

export class Command extends SuperCommand {
  description: string =
    'Takes one optional argument, number of commits to rebase. '
    + 'Commits all staged changes, '
    + 'rebases interactively, '
    + 'amends the most recent commit, '
    + 'then force pushes';
  alias: string = 'ggo';
  public async execute(vital?: boolean, input?: any) {
    await super.executeWithInput(argumentss, options, input, vital, async () => {
      await (new GitCommit()).execute(false);
      if (this.input[Names.NUMBER]) await (new GitRebase()).execute(true, input);
      await (new GitAmend()).execute(true);
      await (new GitForcePush()).execute(true);
    });
  }
}
